src/box/sql/build.c

```diff
@@ -558,7 +552,7 @@ field_def_create_for_pk(struct Parse *parser, struct field_def *field,
 void
 sqlAddPrimaryKey(struct Parse *pParse)
 {
-	struct space *space = pParse->create_column_def.space;
+	struct space *space = pParse->space;
 	assert(space != NULL || pParse->src_list != NULL);
 	if (space == NULL)
 		return sql_create_index(pParse, &pParse->primary_key.name,
```

Comment:
Addition to `struct Parse / struct sql_parse` and naming of all those new `sql_parse_...()` functions. The mega confusing part is that we somewhy use verb as a struct name. Verbs are for function names, not for struct names. Functions like `sql_parse_...` look like they want to "parse" something. Not like methods of an `sql_parse` object.

If you are going to introduce a new struct and you are preparing all those new functions `sql_parse_...` for it, I highly recommend to rethink the naming. Use `sql_ast` or `sql_query` or `sql_stree` (syntax tree) or `sql_ptree` (parser tree) or alike.